### DISCUSSION TITLE
Config File Like Alacritty

### DISCUSSION THREAD

**Discussion by blueray453** [helix-editor/helix#655] [Ideas]
Hi. I use Alacritty. It's config file ( https://github.com/alacritty/alacritty/blob/master/alacritty.yml ) has two features that is extraordinary. 

1. It's config file itself is the documentation. The default values are shown as comments (with `#`). If we want to change, then we uncomment it and change the value. We can also add lines if needed (for example, new keybinding). We do not need to check anywhere else for help (other than `alacritty --help` which show the flags and options of `alacritty` command). It is the single source of truth.

2. Moreover, if we make any change in the config file, the changes reflect in the terminal in real time.

Is it possible to implement something similar.

**Comment by kirawi** [comment]
We have a `config.toml`, but it's fairly barebones, and configuration in general is split across a couple of `.toml` files right now.
https://docs.helix-editor.com/configuration.html

**Comment by oberblastmeister** [comment]
Reload the config on the fly can be implemented using file watching which I made a pr for #588
